features/192-asset-gear/index.md

```diff
@@ -0,0 +1,491 @@
+# Asset Gear
+
+## Overview
+
+The Asset Gear is a gear for uploading and retrieving user-generated asset.
+It supports both public and private asset.
+It can transform image asset on the fly via query string.
+
+## Layout
+
+The Asset Gear uses a single bucket for all apps. The asset name is in the following format.
+
+```
+/<app-id>/<random-string>[.<ext>]
+```
+
+- `app-id`: The app ID.
+- `random-string`: A randomly generated string.
+- `ext`: The inferred file extension (if any) from the given content type.
+
+## HTTP API
+
+### POST /_asset/presign
+
+#### Description
+
+This endpoint returns a presigned upload request which can be used to upload an asset.
+
+This endpoint requires API Key and authenticated user.
+
+#### Request JSON Schema
+
+```json
+{
+  "type": "object",
+  "properties": {
+    "content_type": { "type": "string" },
+    "content_md5": { "type": "string" },
+    "content_length": { "type": "integer" },
+    "access": { "enum": ["public", "private"] }
+  },
+  "required": ["content_type", "content_md5", "content_length"]
+}
+```
+
+- `content_type`: The media type of the asset.
+- `content_md5`: The MD5 hash of the asset.
```

Comment:
Just curious, why is it needed?

Reply:
If content MD5 is provided, the backend storage service (e.g. S3) checks the integrity of the data to ensure the data is not corrupted during upload.

Reply:
Noted thx!